File: distros/A/Algorithm-VSM/examples/corpus_with_java_and_cpp/FirstWindowWithButton.cc
// This code example is from the following source:
//
// Book Title:  Programming with Objects, A Comparative Presentation
//              of Object-Oriented Programming with C++ and Java
//
// Chapter:     Chapter 17  OO For Graphical User Interfaces, A Tour Of Three Toolkits
//
// Section:     Section 17.6  Minimialist GUI Programs In QT
//
// The links to the rest of the code in this book are at
//     
//      http://programming-with-objects.com/pwocode.html
//
// For further information regarding the book, please visit
//
//      http://programming-with-objects.com
//



//FirstWindowWithButton.cc

#include <qapplication.h>
#include <qmainwindow.h>
#include <qpushbutton.h>
#include <qfont.h>
 
int main( int argc, char **argv )
{
    QApplication myApp( argc, argv );                             //(A) 
    QMainWindow* myWin = new QMainWindow( 0, 0, 0 );              //(B)
    myWin->resize( 500, 300 );                                    //(C)
    myWin->move( 200, 100 );                                      //(D)

    QPushButton* quitButton = new QPushButton( "Quit", myWin );   //(E)
    quitButton->resize( 60, 30 );                                 //(F)
    quitButton->move( 220, 135 );                                 //(G)
    quitButton->setFont( QFont( "Times", 18, QFont::Bold ) );     //(H)
 
    QObject::connect( quitButton, 
                      SIGNAL(clicked()), 
                      &myApp, 
                      SLOT(quit()) );                             //(I)
    myApp.setMainWidget( myWin );                                 //(J)
    myWin->show();                                                //(L)
    return myApp.exec();                                          //(M)
}                          

File: distros/A/Algorithm-VSM/examples/corpus_with_java_and_cpp/FirstWindowWithButton.java
// This code example is from the following source:
//
// Book Title:  Programming with Objects, A Comparative Presentation
//              of Object-Oriented Programming with C++ and Java
//
// Chapter:     Chapter 17  OO For Graphical User Interfaces, A Tour Of Three Toolkits
//
// Section:     Section 17.5  Minimalist GUI Programs In AWT/Swing
//
// The links to the rest of the code in this book are at
//     
//      http://programming-with-objects.com/pwocode.html
//
// For further information regarding the book, please visit
//
//      http://programming-with-objects.com
//



//FirstWindowWithButton.java

import javax.swing.*;
import java.awt.*;                   // for FlowLayout
import java.awt.event.*;             // for ActionListener

class FirstWindowWithButton {
    public static void main(String[] args) {
        JFrame f = new JFrame( "FirstWindowWithButton" );   

        f.addWindowListener(new WindowAdapter() {                 //(A)
            public void windowClosing(WindowEvent e) {
                System.exit(0);
            }
        });

        JButton b = new JButton("Click Here for a Message");      //(B)
        b.setVerticalTextPosition( AbstractButton.CENTER );       //(C)
        b.setHorizontalTextPosition( AbstractButton.CENTER );     //(D)

        b.addActionListener( new ActionListener() {               //(E)
            public void actionPerformed( ActionEvent evt ) {
                System.out.println( "Have a Good Day!" );
            }
        });

        f.getContentPane().setLayout( new FlowLayout() );         //(F)
        f.getContentPane().add( b );                              //(G)
        f.setLocation( 100, 50);                                  //(H)
        f.pack();                                                 //(I)
        f.setVisible( true );                                     //(J)
    }
}
